sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -150,6 +151,29 @@ abstract class JdbcDialect extends Serializable {
   def beforeFetch(connection: Connection, properties: Map[String, String]): Unit = {
   }
 
+  /**
+   * Do some extra properties validation work in addition to the validation in [[JDBCOptions]],
+   * eg. [[validateFetchSize()]].
+   * @param properties The connection properties.  This is passed through from the relation.
+   */
+  def validateProperties(properties: Map[String, String]): Unit = {
+    validateFetchSize(properties)
+  }
+  /**
+   * This is to validate the fetch size specified via [[JDBCOptions.JDBC_BATCH_FETCH_SIZE]].
+   * The implementation here requires the fetch size >= 0, sub classes may override it to meet
+   * requirement of different dialects.
+   * @param properties The connection properties.  This is passed through from the relation.
+   */
+  protected def validateFetchSize(properties: Map[String, String]): Unit = {
```

Comment:
Why not just pass the fetch size as the parameter to this method? avoids having to duplicate the logic that gets the value